src/Microsoft.Extensions.ProjectModel.Sources/DotNet/DotNetDependencyProvider.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Compilation;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.ProjectModel.Resolution;
+
+namespace Microsoft.Extensions.ProjectModel
+{
+    internal class DotNetDependencyProvider
+    {
+        private ProjectContext _context;
+        private List<DependencyDescription> _packageDependencies;
+        private List<ResolvedReference> _resolvedReferences;
+        private string _configuration;
+
+        public DotNetDependencyProvider(ProjectContext context, string configuration = "Debug")
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            _configuration = configuration;
+            _context = context;
+            DiscoverDependencies();
+
+        }
+
+        public IEnumerable<DependencyDescription> GetPackageDependencies()
+        {
+            return _packageDependencies;
+        }
+
+        public IEnumerable<ResolvedReference> GetResolvedReferences()
+        {
+            return _resolvedReferences;
+        }
+
+        private void DiscoverDependencies()
+        {
+            LibraryExporter exporter = _context.CreateExporter(_configuration);
+
+            var framework = _context.TargetFramework;
+            if (exporter == null)
+            {
+                throw new ArgumentNullException(nameof(exporter));
+            }
+
+            if (framework == null)
+            {
+                throw new ArgumentNullException(nameof(framework));
```

Comment:
`InvalidOperationException("Couldn't resolve dependencies when target framework is not specified.")`